Source/BestProjectEver/Private/Core/GameModes/BPE_GameplayGameMode.cpp

```diff
@@ -39,7 +40,14 @@ void ABPE_GameplayGameMode::InitializeReferences()
 {
 	TimeLeft = WarmupTime;
 	LevelStartingTime = GetWorld()->GetTimeSeconds();
+	
 	GameStateReference = Cast<ABPE_GameState>(GetWorld()->GetGameState());
+	if(IsValid(GameStateReference))
+	{
+		TArray<AActor*> EnemiesInGame;
+		UGameplayStatics::GetAllActorsOfClass(GetWorld(), ABPE_Enemy::StaticClass(), EnemiesInGame);
+		GameStateReference->SetEnemiesOnMatch(EnemiesInGame.Num());
```

Comment:
Why do this in the game mode if you're using it to set a variable in the game state? Why not do this directly inside the game state?

Reply:
![image](https://user-images.githubusercontent.com/85327912/174116497-63b1edfe-fec1-4e3d-8eba-41290ef18b4d.png)